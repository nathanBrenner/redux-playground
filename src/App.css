.container {
  width: 100%;
  max-width: 600px;
  margin: 0 auto;
  border: 1px solid red;
  padding: 2rem;
}
