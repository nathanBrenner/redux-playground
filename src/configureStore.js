import todoApp from "./reducers";
import { createStore, applyMiddleware } from "redux";
import promise from "redux-promise";
import { createLogger } from "redux-logger";

const configureStore = () => {
  const middlewares = [promise];

  if (process.env.NODE_ENV !== "production") {
    middlewares.push(createLogger());
  }

  const store = createStore(
    todoApp,
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(...middlewares)
  );
  //     window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  return store;
};

export default configureStore;
