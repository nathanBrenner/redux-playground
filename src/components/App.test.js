import React from "react";
import App from "./App";
import configureStore from "../configureStore";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";

describe("App", () => {
  test("renders without crashing", () => {
    const store = configureStore();
    shallow(<App store={store} />);
  });

  test("snapshot", () => {
    const store = configureStore();
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
