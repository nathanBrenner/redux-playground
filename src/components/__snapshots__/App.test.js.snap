// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <div>
    <input />
    <button
      onClick={[Function]}
    >
      Add Todo
    </button>
  </div>
  <p>
    Loading...
  </p>
  <p>
    Show: 
    <a
      aria-current="true"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
      All
    </a>
    , 
    <a
      aria-current={false}
      className={undefined}
      href="/active"
      onClick={[Function]}
      style={undefined}
    >
      Active
    </a>
    , 
    <a
      aria-current={false}
      className={undefined}
      href="/completed"
      onClick={[Function]}
      style={undefined}
    >
      Completed
    </a>
  </p>
</div>
`;
